pylint_nautobot/status_field_instead_of_status_model.py

```diff
@@ -0,0 +1,25 @@
+"""Check for deprecated StatusModel usage and encourage the usage of StatusField instead."""
+from pylint.checkers import BaseChecker
+from pylint.interfaces import IAstroidChecker
+
+
+class NautobotDeprecatedStatusModelChecker(BaseChecker):
+    """Discourage the usage of deprecated StatusModel and encourage the usage of StatusField."""
+
+    __implements__ = IAstroidChecker
+
+    version_specifier = ">=2,<3"
+
+    name = "nautobot-status-field-instead-of-status-model"
+    msgs = {
+        "E4292": (
+            "Inherits from the deprecated StatusModel instead of declaring status on the model explicitly with StatusField",
```

Comment:
Wondering if the preferred phrasing should be more imperative, something like:

```suggestion
            "Declare status on the model explicitly with a StatusField, as StatusModel is deprecated",
```

Reply:
If so, should this rule also enforce users creating manual foreign keys to the `Status` model?

Reply:
Yeah, it wouldn't hurt to have a rule warning about `ForeignKey("extras.Status")` as well. Probably a separate rule though for clarity?

Reply:
Yes, and to not hold things up it doesn't have to be in this PR. #19 tracks this already and should be updated accordingly.

Reply:
The first string is the *error* message itself - so the bad pattern etc. that is detected and should be avoided. Any explanation goes into the thirds string in the tuple (which currently has just a link).

Reply:
Yup, I just looked that up, Cristian is right and the first string should describe the error, see [here](https://pylint.readthedocs.io/en/latest/user_guide/messages/error/abstract-class-instantiated.html) for example.